x-pack/plugins/fleet/server/services/output.ts

```diff
@@ -178,18 +178,39 @@ async function validateTypeChanges(
     // Validate no policy with fleet server use that policy
     validateLogstashOutputNotUsedInFleetServerPolicy(fleetServerPolicies);
   }
-  // if a logstash output is updated to become default, update the fleet server policies to use the previous ES output or default output
-  if (data?.type === outputType.Logstash && mergedIsDefault) {
+  await updateFleetServerPoliciesDataOutputId(
+    soClient,
+    esClient,
+    data,
+    mergedIsDefault,
+    defaultDataOutputId,
+    fleetServerPolicies
+  );
+}
+
+async function updateFleetServerPoliciesDataOutputId(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  data: Partial<Output>,
+  isDefault: boolean,
+  defaultDataOutputId: string | null,
+  fleetServerPolicies: AgentPolicy[]
+) {
+  // if a logstash output is updated to become default
+  // update the fleet server policies to use the previous ES output or default output
+  if (data?.type === outputType.Logstash && isDefault) {
     for (const policy of fleetServerPolicies) {
-      await agentPolicyService.update(
-        soClient,
-        esClient,
-        policy.id,
-        { data_output_id: defaultDataOutputId },
-        {
-          force: true,
-        }
-      );
+      if (!policy.data_output_id && !policy.is_preconfigured) {
```

Comment:
I think the logic may not be totally correct here, the way I saw that we should update all the `fleetServerPolicies` that are using the changed output and not all the `fleetServerPolicies` so the condition should be something like
```
(isDefault && !policy.data_output_id) ||  policy.data_output_id === outputId
```
Does it make sense to you?

Reply:
The `isDefault` flag is related to the output and it's already checked in row `201`, so I think it should be more`!policy.data_output_id || policy.data_output_id === data?.id`. 

Also, do you think that the preconfigured policies should be allowed to be changed here?  I had the check `is_preconfigured` here to avoid changing preconfigured policies in a way that the customer doesn't expect. It'd be better to be more restrictive here. I'd go with 
```
        !policy.data_output_id ||
        !policy.is_preconfigured ||
        policy.data_output_id === data?.id
```

Reply:
I do not think we should have a specific case for `preconfigured` but we may should not pass `force` in the upate call but pass the `fromPreconfiguration` flag? so in that case it will fail if it try to update a preconfigured output and not result in a non valid comment

Reply:
I removed the `!is_preconfigured` check and also the `force` flag in the update call. It should be good now